src/worker/users.js

```diff
@@ -77,37 +85,62 @@ class Users {
             return null;
         }
 
-        if (checkPass && !await user.checkPassword(password)) {
+        if (!isUserToken && !await user.checkPassword(password)) {
             return null;
         }
 
+        if (isUserToken) {
+            this.updateUserTokenAccess(user.id, password, userHost);
+        }
+
         return user;
     }
 
-    async authUserToken(token) {
+    async authUserToken(token, userHost) {
         let user = this.db.dbUsers('users')
             .innerJoin('user_tokens', 'users.id', 'user_tokens.user_id')
             .where('user_tokens.token', token)
             .first()
             .then(this.db.factories.User.fromDbResult);
 
         if (user) {
+            this.updateUserTokenAccess(user.id, token, userHost);
```

Comment:
`userHost` should probably be optional so that it only `updateUserTokenAccess()` if one of given? There may be a time when we want to check a token but not consider it a login auth.